### ISSUE TITLE
Get rid of storyboards

### ISSUE THREAD

**Issue by konnork** [mhacks/MHacks-iOS#26]
## Goals:
* Allow for custom view controller initializers (for dependency injection)
* Prevent difficult merge conflicts
* Easier view reuse

**Comment by manavgabhawala** [comment]
I just want to say, I actually like having storyboards, it makes it convenient and easy to add layout attributes and visualize the app hierarchy. Storyboards are also a great place for new comers to start looking at rather than jumping into some code.
Finally, I always question big changes by asking is the effort and restructuring work required worth it? I think adding new and better features is a far more prudent use of time than trying to get rid of the storyboard.
I know in theory having no storyboards is ideal, but this is just my 2¢.

**Comment by konnork** [comment]
Yeah it is definitely easier for newcomers, especially since most of them (and more as we go) will not know how to /not/ have storyboards.

That said, I find it much cleaner to not use storyboards, and the limitations (like compile time checking) can get annoying. I don't really see tooooo much of a time issue getting rid of it as quite a few view controllers are independent already.

At the very least I'd like to refactor each tab to its own storyboard. I'll ask the iOS peeps at the meeting on Monday what they think
